docs/concepts/automations.md

```diff
@@ -132,6 +90,205 @@ Prefect tries to infer the relevant event whenever possible, but sometimes one d
+
+```json
+"match": {
+  "prefect.resource.id": "prefect.flow-run.*",
+  "prefect.resource.name": ["cute-*", "radical-*"]
+},
+"match_related": {},
+...
+```
+
+This configuration, on the other hand, will filter for any events for which this specific deployment is a related resource.
+
+```json
+"match": {},
+"match_related": {
+  "prefect.resource.id": "prefect.deployment.37ca4a08-e2d9-4628-a310-cc15a323378e"
+},
+...
+```
+
+Both of the above approaches will select the example `prefect.flow-run.Completed` event, but will permit additional, possibly undesired events through the filter as well. `match` and `match_related` can be combined for more restrictive filtering:
+
+```json
+"match": {
+  "prefect.resource.id": "prefect.flow-run.*",
+  "prefect.resource.name": ["cute-*", "radical-*"]
+},
+"match_related": {
+  "prefect.resource.id": "prefect.deployment.37ca4a08-e2d9-4628-a310-cc15a323378e"
+},
+...
+```
+
+Now this trigger will filter only for events whose primary resource is a flow run started by a specific deployment, _and_ that flow run has a name starting with `cute-` or `radical-`.
+
+### Expected events
+
+Once an event has passed through the `match` filters, it must be decided if this event should be counted toward the trigger's `threshold`. Whether that is the case is determined by the event names present in `expect`.
+
+This configuration informs the trigger to evaluate _only_ `prefect.flow-run.Completed` events that have passed the `match` filters.
+
+```json
+"expect": [
+  "prefect.flow-run.Completed"
+],
+...
+```
+
+`threshold` decides the quantity of `expect`ed events needed to satisfy the trigger. Increasing the `threshold` above 1 will also require use of `within` to define a range of time in which multiple events are seen. The following configuration will expect two occurrences of `prefect.flow-run.Completed` within 60 seconds.
+
+```json
+"expect": [
+  "prefect.flow-run.Completed"
+],
+"threshold": 2,
+"within": 60,
+...
+```
+
+`after` can be used to handle scenarios that require more complex event reactivity.
+
+Take, for example, this flow which emits an event indicating the table it operates on is missing or empty:
+
+```python
+from prefect import flow
+from prefect.events import emit_event
+from db import Table
+
+
+@flow
+def transform(table_name: str):
+  table = Table(table_name)
+
+  if not table.exists():
+    emit_event(
+        event="table-missing", resource={"prefect.resource.id": "etl-events.transform"}
+    )
+  elif table.is_empty():
+    emit_event(
+        event="table-empty", resource={"prefect.resource.id": "etl-events.transform"}
+    )
+  else:
+    # transform data
+```
+
+The following configuration uses `after` to prevent this automation from firing unless either a `table-missing` or a `table-empty` event has occurred before a flow run of this deployment completes. Note how `match` and `match_related` are used to ensure the trigger only evaluates events that are relevant to its purpose:
```

Comment:
nit: perhaps this sentence

> Note how `match` and `match_related` are used to ensure the trigger only evaluates events that are relevant to its purpose:

could be moved to a `!!! Tip` below the json snippet so its prettier?